File: AtanAlir.Client/src/types/match.ts
// types/match.ts
export interface Match {
    id: number;
    homeTeam: {
        id: number;
        name: string;
    };
    awayTeam: {
        id: number;
        name: string;
    };
    score: {
        fullTime: {
            home: number | null;
            away: number | null;
        };
        halfTime?: {
            home: number | null;
            away: number | null;
        };
    };
    status: string;  // 'SCHEDULED' | 'TIMED' | 'LIVE' | 'IN_PLAY' | 'PAUSED' | 'FINISHED'
    utcDate: string;
    competition: {
        id: number;
        name: string;
    };
}

export interface MatchesResponse {
    liveMatches: Match[];
    upcomingMatches: Match[];
}

File: AtanAlir.Core/Models/Match.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AtanAlir.Core.Models
{
    public class Match
    {
        public int Id { get; set; }
        public string HomeTeam { get; set; } = null!;
        public string AwayTeam { get; set; } = null!;
        public MatchScore? Score { get; set; }
        public string Status { get; set; } = null!;
        public DateTime StartTime { get; set; }
    }

    public class MatchScore
    {
        public int Home { get; set; }
        public int Away { get; set; }
    }

    public class MatchesResponse
    {
        public List<Match> LiveMatches { get; set; } = new();
        public List<Match> UpcomingMatches { get; set; } = new();
    }
}
